scss/util/_color.scss

```diff
@@ -92,35 +107,39 @@
   @return scale-color($color, $lightness: $scale);
 }
 
-/// Get color from foundation-palette
+/// Get color from foundation-colors
 ///
-/// @param {key} color key from foundation-palette
+/// @param {key} color key from foundation-colors
 ///
-/// @returns {Color} color from foundation-palette
+/// @returns {Color} color from foundation-colors
 @function get-color($key) {
-  @if map-has-key($foundation-palette, $key) {
-    @return map-get($foundation-palette, $key);
-  }
-  @else {
-    @error 'given $key is not available in $foundation-palette';
-  }
+  @return map-safe-get($foundation-colors, $key);
+}
+
+/// Get color state from foundation-states
+///
+/// @param {key} state key from foundation-states
+///
+/// @returns {Color} color from foundation-states
+@function get-state($key) {
```

Comment:
I would suggest to make this internal `-zf-get-state` and wrap generation of states for components. Otherwise I'm sure this will cause a lot of confusion why `get-color(green)` is equal to `get-state(primary)` and why both returns a color. Or it should be `get-state-color` to be sure, that nobody expects something else than a color.